### ISSUE TITLE
Bundle the boost git repo instead of using the .tar.gz archiv

### ISSUE THREAD

**Issue by lenaploetzke** [SciCompMod/memilio#983]
### Motivation / Current Behaviour

We currently bundle only a minimal extract of boost library as an archive that contains only the libraries of boost that we use. E.g. in PR #929 we need additional functionality from boost. The resulting bundle of boost is too big to include it in the repository.

### Enhancement description

The preferred solution is to bundle the git repo of boost instead of using the .tar.gz archive. A disadvantage is that the build step of the project takes a while because boost is downloaded. 

### Checklist

- [X] Attached labels, especially loc:: or model:: labels.
- [X] Linked to project

**Comment by xsaschako** [comment]
I already tried this. As far as i know, there is no elegant way to do this.
Do you have an idea in mind?
There currently is minimal CMake support for boost for this but to use it i think it is not ready yet. Last time i looked into it was 1/2 year ago.

**Comment by lenaploetzke** [comment]
Yes, I have a few ideas and some things have already worked out. I've also talked to @dabele. Maybe I'm missing something that's causing the whole thing to fail. I'll try a few things. Will keep you updated.
